Cabal/doc/developing-packages.rst

```diff
@@ -2063,21 +2063,31 @@ system-dependent values for these fields.
 .. pkg-field:: build-tool-depends: package:executable list
     :since: 2.0
 
-    A list of Haskell programs needed to build this component.
+    A list of Haskell executables needed to build this component. Executables are provided during the whole duration of the build,
+    so this field can be used for executables needed during :pkg-section:`test-suite` as well.
+
     Each is specified by the package containing the executable and the name of the executable itself, separated by a colon, and optionally followed by a version bound.
-    It is fine for the package to be the current one, in which case this is termed an *internal*, rather than *external* executable dependency.
 
-    External dependencies can (and should) contain a version bound like conventional :pkg-field:`build-depends` dependencies.
-    Internal deps should not contain a version bound, as they will be always resolved within the same configuration of the package in the build plan.
-    Specifically, version bounds that include the package's version will be warned for being extraneous, and version bounds that exclude the package's version will raise an error for being impossible to follow.
+    All executables defined in the given Cabal file are termed as *internal* dependencies as opposed to the rest which are *external* dependencies.
+    Each of the two is handled differently:
+
+    1. External dependencies can (and should) contain a version bound like conventional :pkg-field:`build-depends` dependencies.
+    2. Internal depenedencies should not contain a version bound, as they will be always resolved within the same configuration of the package in the build plan.
+       Specifically, version bounds that include the package's version will be warned for being extraneous, and version bounds that exclude the package's version will raise an error for being impossible to follow.
 
-    Cabal can make sure that specified programs are built and on the ``PATH`` before building the component in question.
-    It will always do so for internal dependencies, and also do so for external dependencies when using Nix-style local builds.
+    Cabal tries to make sure that all specified programs are built and provided on the ``PATH`` before building the component in question, but can only do so for Nix-style builds. Specifically:
```

Comment:
One thing worth mentioning explicitly is that those tools' are prepended to the search `$PATH`; that's important as otherwise a globally same-called executable could interfere (concrete example: if you happen to have a system-wide installed `alex` executable version 3.1.3 in `/usr/bin`, but your `build-tools-depends: alex:alex ^>= 3.2.4` you'd certainly want to use the one provisioned by `new-build` satisfying the version constraint to take precedence.

Reply:
"are built" ... does this mean `cabal` will do the building automatically or that the user is expected to do the building independently?

Reply:
@fommil would "are atomically built" clarify the intent?